File: math/prime-sum.cpp
bool isPrime(int k){
    for(int i=2; i<=sqrt(k); i++){
        if(k%i==0) return false;
    }
    return true;
}
vector<int> Solution::primesum(int A) {
    vector<int> ans;
    for(int i=2; i<=A/2; i++){
        if(isPrime(i)){
            if(isPrime(A-i)){
                ans.push_back(i);
                ans.push_back(A-i);
                return ans;
            }
        }
    }
    return ans;
}

File: math/prime-sum.py
class Solution:
    primes = [False, False]

    def cal_primes(self, A):
        length = len(self.primes)
        self.primes += [True]*(A-length)

        for i in range(2, int(A**0.5)+1):
            if self.primes[i] == True:
                for j in range((length//i)*i, A, i):
                    if i == j: continue
                    self.primes[j] = False

    def primesum(self, A):
        if len(self.primes) < A: self.cal_primes(A)

        for i in range(2, len(self.primes)//2 + 1):
            if self.primes[i]:
                if self.primes[A-i]: return [i, A-i]

